toolkit/tools/srpmpacker/srpmpacker.go

```diff
@@ -758,10 +765,18 @@ func packSingleSPEC(specFile, srpmFile, signaturesFile, buildDir, outDir, distTa
 		return
 	}
 
-	// Hydrate all sources. Download any missing ones not in `sourceDir`
-	err = hydrateFiles(fileTypeSource, specFile, workingDir, srcConfig, currentSignatures, defines)
-	if err != nil {
-		return
+	{
+		// Limit the number of concurrent network operations by pushing a struct{} into the channel. This will block until
+		// another operation completes and removes the struct{} from the channel.
+		netOpsSemaphor <- struct{}{}
+		// Hydrate all sources. Download any missing ones not in `sourceDir`
+		err = hydrateFiles(fileTypeSource, specFile, workingDir, srcConfig, currentSignatures, defines)
+		if err != nil {
+			<-netOpsSemaphor
+			return
+		}
+		// Clear the channel to allow another operation to start
+		<-netOpsSemaphor
```

Comment:
What do you think about moving acquiring the semaphore and releasing it into `hydrateFiles`? This way we can use `defer` to include the release logic only once.

Reply:
I think we might even move it deeper into `hydrateFromRemoteSource` to block right before we call `network.DownloadFile`.

Reply:
Or maybe even move the whole limitation concept into the `network` module, so callers don't even have to think about it. Probably an overkill for now, but something to think about.

Reply:
Done